### PR TITLE
Implement DimensionService.getDataDimensionalItemObjects

### PR DESCRIPTION
**Pull request by jimgrace**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemId.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    // -------------------------------------------------------------------------
+    // hashCode, equals and toString
+    // -------------------------------------------------------------------------
+
+    @Override
+    public boolean equals( Object o )
+    {
+        if ( this == o )
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemId.java]: Codestyle, add brackets { }

Comment (jimgrace) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemId.java]: Brackets added, thanks.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemId.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        switch ( dimensionItemType )
+        {
+            case DATA_ELEMENT:
+            case PROGRAM_INDICATOR:
+                return ids.length == 1 && ids[0] != null;
+
+            case DATA_ELEMENT_OPERAND:
+                return ids.length == 3 && ids[0] != null;
+
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemId.java]: Operand: Could it legally have two identifiers, e.g. <dataelement-id>.<catoptioncombo-id> , i.e. without the attribute option combo id? Perhaps length >= 2 ?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dimension/DefaultDimensionService.java [hunk 3]
```diff
@@ -386,10 +372,218 @@ else if ( !idScheme.is( IdentifiableProperty.UID ) || CodeGenerator.isValidUid(
...
     }
 
+    @Override
+    public DimensionalItemObject getDataDimensionalItemObject( DimensionalItemId itemId )
+    {
+        Map<DimensionalItemId, DimensionalItemObject> objectMap = getDataDimensionalItemObjects(
+            Sets.newHashSet( itemId ) );
+
+        if ( objectMap.values().size() != 0 )
+        {
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dimension/DefaultDimensionService.java]: Use .isEmpty() instead of size() == 0

**Pull request closed without merging**